client/src/services/informationManager.ts

```diff
@@ -0,0 +1,132 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+import { MsAlertModal, MsAlertModalConfig, MsReportTheme } from '@/components/core';
+import { NotificationManager } from '@/services/notificationManager';
+import { ToastManager } from '@/services/toastManager';
+import { modalController } from '@ionic/vue';
+import { v4 as uuid4 } from 'uuid';
+
+export const InformationKey = 'information';
+
+export enum InformationLevel {
+  Info = 'INFO',
+  Success = 'SUCCESS',
+  Warning = 'WARNING',
+  Error = 'ERROR',
+}
+
+export interface InformationOptions {
+  message: string;
+  level: InformationLevel;
+  data?: object;
+}
+
+export class Information {
+  id: string;
+  message: string;
+  level: InformationLevel;
+  data?: object;
+  constructor({ message, level, data = {} }: InformationOptions) {
+    this.id = uuid4();
+    this.message = message;
+    this.level = level;
+    this.data = data;
+  }
+
+  get theme(): MsReportTheme {
+    switch (this.level) {
+      case InformationLevel.Info:
+        return MsReportTheme.Info;
+      case InformationLevel.Success:
+        return MsReportTheme.Success;
+      case InformationLevel.Warning:
+        return MsReportTheme.Warning;
+      case InformationLevel.Error:
+        return MsReportTheme.Error;
+    }
+  }
+}
+
+export enum InformationPresentationMode {
+  Console = 1 << 0,
+  Notification = 1 << 1,
+  Toast = 1 << 2,
+  Modal = 1 << 3,
+}
+
+export class InformationManager {
+  toastManager: ToastManager;
+  notificationManager: NotificationManager;
+
+  constructor() {
+    this.toastManager = new ToastManager();
+    this.notificationManager = new NotificationManager();
+  }
+
+  private async showModal(information: Information): Promise<void> {
+    const alertModalConfig: MsAlertModalConfig = {
+      theme: information.theme,
+      message: information.message,
+    };
+
+    await this._createAndPresentModal(alertModalConfig);
+  }
+
+  private async _createAndPresentModal(modalConfig: MsAlertModalConfig): Promise<any> {
+    const top = await modalController.getTop();
+    if (top) {
+      top.classList.add('overlapped-modal');
+    }
+
+    const modal = await modalController.create({
+      component: MsAlertModal,
+      canDismiss: true,
+      cssClass: 'information-modal',
+      componentProps: modalConfig,
+    });
+    await modal.present();
+    const result = await modal.onWillDismiss();
+    await modal.dismiss();
+
+    if (top) {
+      top.classList.remove('overlapped-modal');
+    }
+    return result;
+  }
+
+  private async showToast(information: Information): Promise<void> {
+    this.toastManager.createAndPresent({
+      theme: information.theme,
+      message: information.message,
+    });
+  }
+
+  private sendNotification(information: Information): void {
+    this.notificationManager.add(information);
+  }
+
+  async present(information: Information, presentationMode: InformationPresentationMode): Promise<void> {
```

Comment:
`presentationMode` should by typed as a `number` (it's a combination of values from InformationPresentationMode)